### ISSUE TITLE
Be able to ignore method in interface

### ISSUE THREAD

**Issue by davz13** [jmattheis/goverter#171]
**Is your feature request related to a problem? Please describe.**
I have a source that has a field []byte that will be unmarshaled and used to populate a few fields on the target. Since I don't think this is possible with the current generator, I want to define my own method to do this, but I want to be able to include that custom method in the converter interface for code organization

**Describe the solution you'd like**
`// goverter:ignore:method `
or just
`// goverter:ignore`
would skip the method from being generated. This would generate an incomplete Implementation.

**Describe alternatives you've considered**
My issue could also perhaps be solved with this #169. I would be able to call the convert with an already initialized object.

Or providing support for custom unflattening of fields
`// goverter:map jsonObjString . | unmarshalJson`

**Comment by jmattheis** [comment]
Could you give an concrete example with structs, the unmarshaljson method, and the conversion interface?

**Comment by davz13** [comment]
```
type InternalModel struct {
	Id            string    `json:"-"`
        Name     string     `json:"name"`  
	Type           string `json:"type"`
}

type DatabaseModel struct {
	Id               string
        JsonString  string
}
```
With the conversion interface
```
func deserialize(input *DatabaseModel, output *InternalModel) {
	jsoniter.Unmarshal(input.JsonString, &output)
}

type Converter interface {
 	// goverter:map . . | deserialize
        convert(input *DatabaseModel, output *InternalModel)
}
```
or
````
func deserialize(input JsonString, output *InternalModel) {
	jsoniter.Unmarshal(input.JsonString, &output)
}

type Converter interface {
 	// goverter:map JsonString . | deserialize
        convert(input *DatabaseModel, output *InternalModel)
}
```
Would be ideal.

If you wanted to generate an incomplete impl, this could work.

````

type Converter interface {

 	// goverter:ignore:method
        convert(input *DatabaseModel, output *InternalModel)

        //goverter:update output
       //ignore Name Type
       partialConvert(input *DatabaseModel, output *InternalModel)
}
```

**Comment by jmattheis** [comment]
If you could change the signature, so that InternalModel is returned instead, then you could do it like this:
```golang
// goverter:converter
type Converter interface {
	// goverter:default Deserialize
	// goverter:ignore Name Type
	Convert(source *DatabaseModel) *InternalModel
}

func Deserialize(input *DatabaseModel) *InternalModel {
	output := &InternalModel{}
	jsoniter.Unmarshal([]byte(input.JsonString), &output)
	return output
}
```

---

Have you considered using another output format, so that you don't have the struct? https://goverter.jmattheis.de/guide/format

---

I do think your current solution with the additional interface fits quite nicely, goverter create a struct matching the interface, and you can define another one and satisfy it.

**Comment by davz13** [comment]
Ahh, missed the default setting. That will work for what I'm trying to do. Thanks!

**Comment by davz13** [comment]
Though I think it'd be helpful to clarify in the docs what kind of methods default can work with. The guide says any optional source will work, but no parameters are supplied in your example https://goverter.jmattheis.de/reference/default

It didn't seem like the target could be constructed using the source.

**Comment by davz13** [comment]
After attempting to use default, I found that it doesn't play nicely with map. I tried with your example:

```
// goverter:converter
type Converter interface {
	// goverter:default NewOutput
	// goverter:map Age Age
	Convert(*Input) *Output
}
```

```
func (c *ConverterImpl) Convert(source *default1.Input) *default1.Output {
	pExampleOutput := default1.NewOutput()
	if source != nil {
		var exampleOutput default1.Output
		exampleOutput.Age = (*source).Age
		if (*source).Name != nil {
			xstring := *(*source).Name
			exampleOutput.Name = &xstring
		}
		pExampleOutput = &exampleOutput
	}
	return pExampleOutput
}
```

**Comment by davz13** [comment]
Actually it looks like it happens already with no other comments: https://github.com/jmattheis/goverter/blob/main/example/default/generated/generated.go

**Comment by jmattheis** [comment]
Ahh I just saw this. I only looked at the PR before. It's about that it overriding the complete default value in case the source type is not nil. Yeah, sorry I gave the wrong suggestion, didn't remember the pointer edge-case there.

So it basically currently would work, both the source and target type wouldn't be a pointer. This would then generate
```go
type Converter interface {
	// goverter:default NewOutput
	Convert(Input) Output
}

func NewOutput() Output {
	name := "jmattheis"
	return Output{Age: 42, Name: &name}
}

// ...

func (c *ConverterImpl) Convert(source default1.Input) default1.Output {
	exampleOutput := default1.NewOutput()
	exampleOutput.Age = source.Age
	if source.Name != nil {
		xstring := *source.Name
		exampleOutput.Name = &xstring
	}
	return exampleOutput
}
```
and with ignore Age it would be 
```go
func (c *ConverterImpl) Convert(source default1.Input) default1.Output {
	exampleOutput := default1.NewOutput()
	if source.Name != nil {
		xstring := *source.Name
		exampleOutput.Name = &xstring
	}
	return exampleOutput
}
```

Let me think about this, I think the current behavior is wanted in some cases, so the new "also update the existing default value in case of pointers" will be behind a config flag like `goverter:default:update`.

---

> [..] The guide says any optional source will work [..] https://goverter.jmattheis.de/reference/default

Having an example for everything is quite a lot of effort. In the default reference you linked this page is referenced https://goverter.jmattheis.de/reference/signature#signature-optional-source which basically list the possible signatures.

**Comment by davz13** [comment]
> Having an example for everything is quite a lot of effort. In the default reference you linked this page is referenced https://goverter.jmattheis.de/reference/signature#signature-optional-source which basically list the possible signatures.

Yes, I understand now. I was trying to see if I there were other params that could be passed in but I think the only two accepted signatures for default tag would be the following, is that correct?
```func NewOutput() Output

func NewOutput(input Input) Output```

**Comment by davz13** [comment]
got it, I'll try with the non pointer converter for now, thanks!

**Comment by jmattheis** [comment]
@davz13 Could you try out the `default-update` branch:

- `go install github.com/jmattheis/goverter/cmd/goverter@default-update`
- or `go run github.com/jmattheis/goverter/cmd/goverter@default-update [pattern]`
- or `go get github.com/jmattheis/goverter@default-update`

This should fix both the default methods for conversions from *T -> *T and T -> *T. This change now requires a value from a default method to be not nil.

**Comment by jmattheis** [comment]
Should be fixed with with [v1.7.0](https://goverter.jmattheis.de/changelog) when [`defaut:update`](https://goverter.jmattheis.de/reference/default#default-update-yes-no) is enabled.

**Comment by davz13** [comment]
works perfectly, thank you!
